eureka-client/src/main/java/com/netflix/discovery/shared/resolver/ResolverUtils.java

```diff
@@ -41,13 +41,40 @@
 
     private static final Logger logger = LoggerFactory.getLogger(ResolverUtils.class);
 
-    private static final String LOCAL_IPV4_ADDRESS = SystemUtil.getServerIPv4();
-
     private static final Pattern ZONE_RE = Pattern.compile("(txt\\.)?([^.]+).*");
 
+    private static Randomizer RANDOMIZER = new Randomizer() {
```

Comment:
ResolverUtils is a utility class and should have no state.  The static setter also makes it impossible to test different randomizers in parallel without forking the JVM.  While that may be an unlikely scenario we prefer to stick to this philosophy throughout the codebase.